src/Components/TasksPage/RunTaskModal.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useContext, useState } from 'react';
+import { RegistryContext } from '../../store';
+import { Alert, Button, Flex, FlexItem, Modal, Text, TextContent, TextVariants } from '@patternfly/react-core';
+import { AVAILABLE_TASKS_ROOT, INFO_ALERT_SYSTEMS, TASKS_API_ROOT } from '../../Helpers/constants';
+import { ERROR_TASK_NOTIFICATION, EXECUTE_TASK_NOTIFICATION, isError } from '../../Helpers/Helpers';
+import { executeTask } from '../../api';
+import InventoryPage from '../InventoryPage/InventoryPage';
+
+export const RunTaskModal = ({ isOpen, setIsOpen, title, description, slug }) => {
+  const [selectedSystems, setSelectedSystems] = useState([] as string[]);
+  const [isExecuting, setIsExecuting] = useState(false);
+  const { getRegistry } = useContext(RegistryContext);
+
+  const submitTask = async (selectedSystems) => {
+    setIsExecuting(true);
+
+    const store = getRegistry().getStore();
+    const result = await executeTask({ task: slug, hosts: selectedSystems });
+    if (isError(result)) {
+      ERROR_TASK_NOTIFICATION(store, result.message);
+    } else {
+      EXECUTE_TASK_NOTIFICATION(store, title, selectedSystems, result.data.id);
+    }
+
+    setIsExecuting(false);
+    setIsOpen(false);
+  };
+
+  return (
+    <Modal
+      title={title}
+      isOpen={isOpen}
+      onClose={() => setIsOpen(false)}
+      width={'70%'}
+      actions={[
+        <Button
```

Comment:
After this is clicked it would be nice to refresh the table :) (can be separate task in separate PR)

Reply:
Yep, It is planned :+1: